cli/src/pcluster/config/cluster_config.py

```diff
@@ -1755,17 +1874,36 @@ def queue_ami(self):
 
 
 class SlurmQueue(_CommonQueue):
-    """Represent the Slurm Queue resource."""
+    """Represents a Slurm Queue that has Compute Resources with both Single and Multiple Instance Types."""
 
     def __init__(
         self,
-        compute_resources: List[SlurmComputeResource],
+        compute_resources: List[_BaseSlurmComputeResource],
         networking: SlurmQueueNetworking,
+        allocation_strategy: str = "lowest-price",
         **kwargs,
     ):
         super().__init__(**kwargs)
         self.compute_resources = compute_resources
         self.networking = networking
+        if any(isinstance(compute_resource, SlurmFlexibleComputeResource) for compute_resource in compute_resources):
+            self.allocation_strategy = allocation_strategy
+
+    @property
+    def instance_type_list(self):
+        """Return the list of instance types associated to the Queue."""
+        instance_types = set()
+        for compute_resource in self.compute_resources:
+            instance_types.update(compute_resource.instance_types)
+        return list(instance_types)
+
+    @property
+    def instance_types_with_instance_storage(self):
+        """Return a set of instance types in the queue that have instance store."""
+        result = set()
+        for compute_resource in self.compute_resources:
```

Comment:
`compute_resource.instance_types_with_instance_storage` returns a **list** of instance_types so set comprehension won't work since lists are unhashable.